### PR TITLE
Remove locking system APIs

### PR DESCRIPTION
**Pull request by jvff**
## Motivation

PR #1437 removed support for reentrant calls, so locking application state is no longer necessary.

## Proposal

Remove all locking APIs from the system APIs.

## Test Plan

Removed some tests specific to locking, and nothing new needs to be covered.

## Release Plan

Changes the system APIs in the WIT interface, as well as parts of the SDK, so
- Need to bump the major/minor version number in the next release of the crates.
- Need to update the developer manual.

## Links

- [reviewer checklist](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md#reviewer-checklist)

This is part of #1152, but it does **not** close it.

### CODE REVIEW

File: linera-execution/src/runtime.rs [hunk 1]
```diff
@@ -617,54 +592,23 @@ impl<const W: bool> BaseRuntime for SyncRuntimeInternal<W> {
...
-    }
-
-    fn lock_wait(&mut self, _promise: &Self::Lock) -> Result<(), ExecutionError> {
-        Ok(())
-    }
-
-    fn unlock_new(&mut self) -> Result<Self::Unlock, ExecutionError> {
-        let id = self.application_id()?;
-        let state = self.view_user_states.entry(id).or_default();
-        ensure!(state.locked, ExecutionError::ApplicationStateNotLocked(id));
```
Comment (ma2bd) [inline linera-execution/src/runtime.rs:636]: Have you looked at removing the now-unused error variants?

**Review by ma2bd — APPROVED** [review]
Need to remove some error variants and update expected gas values in the test but otherwise LGTM.

File: linera-execution/src/runtime.rs (hunk 1 above)
Comment (jvff) [reply linera-execution/src/runtime.rs:636]: Oops! Fixed :sweat_smile: Thanks!

File: linera-execution/tests/wasm.rs [hunk 2]
```diff
@@ -26,10 +26,10 @@ use test_case::test_case;
 /// called correctly and consume the expected amount of fuel.
 ///
 /// To update the bytecode files, run `linera-execution/update_wasm_fixtures.sh`.
-#[cfg_attr(feature = "wasmer", test_case(WasmRuntime::Wasmer, 28_496, ExecutionRuntimeConfig::Synchronous; "wasmer"))]
-#[cfg_attr(feature = "wasmer", test_case(WasmRuntime::WasmerWithSanitizer, 28_785, ExecutionRuntimeConfig::Synchronous; "wasmer_with_sanitizer"))]
-#[cfg_attr(feature = "wasmtime", test_case(WasmRuntime::Wasmtime, 28_785, ExecutionRuntimeConfig::Synchronous; "wasmtime"))]
-#[cfg_attr(feature = "wasmtime", test_case(WasmRuntime::WasmtimeWithSanitizer, 28_785, ExecutionRuntimeConfig::Synchronous; "wasmtime_with_sanitizer"))]
+#[cfg_attr(feature = "wasmer", test_case(WasmRuntime::Wasmer, 29_140, ExecutionRuntimeConfig::Synchronous; "wasmer"))]
```
Comment (ma2bd) [inline linera-execution/tests/wasm.rs:29]: Fuel consumption is going up??

Comment (jvff) [reply linera-execution/tests/wasm.rs:29]: I found that weird as well. But I can't rule out that it's just a compilation issue. I can't remember if I was the last one to update the fixtures, so someone else might have gotten a more optimized binary. I can run a before and after test and report back.

Comment (jvff) [reply linera-execution/tests/wasm.rs:29]: Okay, I updated the test fixtures with the commit on `main` before this PR was merged, and I got the fuel consumption values of 29,152 for Wasmer without being sanitized and 29,457 for all others. So with this PR there's a slight improvement of 12 fuel units to 29,140 and 29,445 respectively.

**Pull request merged**